TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Migration/Transformations/CreateNodes.php

```diff
@@ -0,0 +1,188 @@
+ * Else it would create the number of identical nodes specified by `amount` setting, defaults to 1.
+ *
+ * `path` specifies on what path relative to current node should the new nodes be created, defaults to the node itself.
+ * `type` specifies the type of nodes to be created.
+ * `properties` array specifies the default properties for the created nodes.
+ */
+class CreateNodes extends AbstractTransformation
+{
+    /**
+     * @Flow\Inject
+     * @var NodeFactory
+     */
+    protected $nodeFactory;
+
+    /**
+     * @Flow\Inject
+     * @var NodeTypeManager
+     */
+    protected $nodeTypeManager;
+
+    /**
+     * @var string
+     */
+    protected $amount = 1;
+
+    /**
+     * @var string
+     */
+    protected $path;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var array
+     */
+    protected $properties;
+
+    /**
+     * @var string
+     */
+    protected $dynamicProperty;
+
+    /**
+     * @var array
+     */
+    protected $dynamicPropertySource;
+
+    /**
+     * Sets the amount of nodes to create.
+     *
+     * @param string $amount
+     * @return void
+     */
+    public function setAmount($amount)
+    {
+        $this->amount = $amount;
+    }
+
+    /**
+     * Sets the path on which to create the node, relative to current node.
+     *
+     * @param string $path
+     * @return void
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+    }
+
+    /**
+     * Sets the type of nodes to be created.
+     *
+     * @param string $type
+     * @return void
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * Sets default values for properties of nodes to be created.
+     *
+     * @param array $properties
+     * @return void
+     */
+    public function setProperties($properties)
+    {
+        $this->properties = $properties;
+    }
+
+    /**
+     * Sets a value of dynamic property, that would be filled from dynamicPropertySource.
+     *
+     * @param string $dynamicProperty
+     * @return void
+     */
+    public function setDynamicProperty($dynamicProperty)
+    {
+        $this->dynamicProperty = $dynamicProperty;
+    }
+
+    /**
+     * Sets the array of values from which dynamicProperty is filled.
+     *
+     * @param array $dynamicPropertySource
+     * @return void
+     */
+    public function setDynamicPropertySource($dynamicPropertySource)
+    {
+        $this->dynamicPropertySource = $dynamicPropertySource;
+    }
+
+    /**
+     * If the given node has the property this transformation should work on, this
+     * returns TRUE.
+     *
+     * @param NodeData $node
+     * @return boolean
+     */
+    public function isTransformable(NodeData $node)
+    {
+        return ($node instanceof NodeData);
```

Comment:
If the method can be called, it must be a NodeData instance (type hint, eh?), so no need to check anything.